src/components/plots/helpers/heatmap/utils.js

```diff
@@ -0,0 +1,38 @@
+const hexToRgb = (hex) => {
+  if (hex) {
+    const i = parseInt(hex.replace(/^#/, ''), 16);
+    const r = (i >> 16) & 255;
+    const g = (i >> 8) & 255;
+    const b = i & 255;
+    return [r, g, b];
+  }
+  return null;
+};
+
+const convertRange = (value, r1, r2) => {
+  // prevent devision by zero
+  if (r1[0] === r1[1]) return value;
+
+  // eslint-disable-next-line no-mixed-operators
+  return (value - r1[0]) * (r2[1] - r2[0]) / (r1[1] - r1[0]) + r2[0];
+};
+
+const listToMatrix = (list, elementsPerSubArray) => {
+  const matrix = []; let i; let
+    k;
+
+  // eslint-disable-next-line no-plusplus
+  for (i = 0, k = -1; i < list.length; i++) {
+    if (i % elementsPerSubArray === 0) {
+      // eslint-disable-next-line no-plusplus
+      k++;
+      matrix[k] = [];
+    }
+
+    matrix[k].push(list[i]);
+  }
+
+  return matrix;
+};
```

Comment:
Why writing this function in this way? I know it works, but it is different style from the rest of the code in the ui.

Reply:
yep, I'll review it

Reply:
After some refactoring `listToMatrix` wasn't necessary anymore, so I removed the function